HST/finalize_data_product.py

```diff
@@ -0,0 +1,139 @@
+##########################################################################################
+# finalize_data_product/__init__.py
+#
+# - Move data directory from staging to bundles directory.
+# - Create data csv.
+# - Create data xml label.
+##########################################################################################
+
+import datetime
+import os
+import pdslogger
+
+from hst_helper import COL_NAME_PREFIX
+from hst_helper.fs_utils import (get_formatted_proposal_id,
+                                 get_format_term,
+                                 get_deliverable_path)
+from hst_helper.general_utils import (create_collection_label,
+                                      create_csv,
+                                      date_time_to_date,
+                                      get_citation_info,
+                                      get_collection_label_data,
+                                      get_mod_history_from_label)
+from product_labels.suffix_info import (INSTRUMENT_NAMES,
+                                        get_collection_title_fmt)
+
+COL_DATA_LABEL_TEMPLATE = 'PRODUCT_COLLECTION_LABEL.xml'
+
+def label_hst_data_directory(proposal_id, logger):
+    """With a given proposal id, move data directory in the final bundle.
+    1. Move data directory from staging to bundles directory.
+    2. Create data csv.
+    3. create data xml label.
+
+    Inputs:
+        proposal_id:    a proposal id.
+        logger:         pdslogger to use; None for default EasyLogger.
+    """
+    logger = logger or pdslogger.EasyLogger()
+    logger.info(f'Label hst data directory with proposal id: {proposal_id}')
+    try:
+        proposal_id = int(proposal_id)
+    except ValueError:
+        logger.exception(ValueError)
+        raise ValueError(f'Proposal id: {proposal_id} is not valid.')
+
+    # Collect data to construct data dictionary used for the labels
+    deliverable_path = get_deliverable_path(proposal_id)
+    for dir in os.listdir(deliverable_path):
+        for col_name in COL_NAME_PREFIX:
+            if dir.startswith(col_name): # work on data_directory
+                col_name = dir
```

Comment:
It's not a great idea to change the variable of a for loop, because someone might try to use the original value later and cause a bug. I would name this variable something else.